crates/sui-core/src/consensus_manager/mod.rs

```diff
@@ -201,26 +127,11 @@ impl ConsensusManagerTrait for ConsensusManager {
     ) {
         let protocol_manager = {
             let mut active = self.active.lock();
-            active.iter().enumerate().for_each(|(index, active)| {
-                assert!(
-                    !*active,
-                    "Cannot start consensus. ConsensusManager protocol {index} is already running"
-                );
-            });
-            let protocol = Self::get_consensus_protocol_in_epoch(&epoch_store);
-            info!("Starting consensus protocol {protocol:?} ...");
-            match protocol {
-                ConsensusProtocol::Narwhal => {
-                    active[0] = true;
-                    self.consensus_client.set(self.narwhal_client.clone());
-                    &self.narwhal_manager
-                }
-                ConsensusProtocol::Mysticeti => {
-                    active[1] = true;
-                    self.consensus_client.set(self.mysticeti_client.clone());
-                    &self.mysticeti_manager
-                }
-            }
+            assert!(!*active, "Cannot start consensus. It is already running!");
+            info!("Starting consensus ...");
+            *active = true;
+            self.consensus_client.set(self.mysticeti_client.clone());
```

Comment:
Do we need to do this on start? Can we just set consensus client when we create ConsensusManager and then we dont need mysticeti_client and consensus_client to be a part of the struct

Reply:
Before we decide on that we will never have a mode switching consensus between epoch, leaving the logic seems bettter.